File: 2trabajoConstructores/js/clase abstracta/AnimalSinParametros.js
const Animal = require("./Animal.js");

class AnimalSinParametros extends Animal {
    constructor() {
        super();
        this.nombre = "Animal genérico";
        this.sonido = "sonido desconocido";
    }

    hacerSonido() {
        console.log(`${this.nombre} hace: ${this.sonido}`);
    }
}

module.exports = AnimalSinParametros;

File: 2trabajoConstructores/c#/claseAbstracta/AnimalSinParametros.cs
public class AnimalSinParametros : AnimalBase
{
    public AnimalSinParametros()
    {
        Nombre = "Desconocido";
        Sonido = "N/A";
    }

    public override void HacerSonido()
    {
        Console.WriteLine($"{Nombre} hace: {Sonido}");
    }
}
